Tasks/AzureResourceGroupDeploymentV2/task.json

```diff
@@ -162,14 +162,23 @@
                 "editorExtension": "ms.vss-services-azure.azurerg-parameters-grid"
             }
         },
+        {
+            "name": "deploymentName",
+            "type": "string",
+            "label": "Deployment name",
+            "defaultValue": "",
+            "required": false,
+            "groupName": "Template",
```

Comment:
@RoopeshNair should this be part of "AdvancedDeploymentOptions" group?

Reply:
We don't have advanced options for AzureResourceGroup deployment task as of yet. There is one for configuring VMs. Introducing a new section just for the deployment name doesn't look good. 
One option could be to rename `Output` section to `Advanced option` and have both `Deployment name` and `Deployment output` under it. @pulkitaggarwl

Reply:
+ 1 to "rename Output section to Advanced option and have both Deployment name and Deployment output under it."
Please update the readme with these details. Also put validations on DeploymentName fields which needs to be adhere to, so that Deployment Name meets any required format.

Reply:
@pulkitaggarwl @RoopeshNair Do not you think that renaming the `Outputs` section to `Advanced deployment options` would be better?

Thus, we would have both options `Advanced deployment options` and `Advanced deployment options for virtual machines`.

Reply:
Sorry, but I am not able to comprehend the rationale behind renaming `Outputs` section to `Advanced Deployment Options`. `Deployment outputs` is an output and not an option! If anything we should move it to `Output variables` section.
I would rather agree to the original change in the PR - with small change of making `Deployment name` input as last one in the main section.

Reply:
@bishal-pdMSFT I've had the same feeling about this one when asked to rename the section. But the first comment also brought the idea of having a `Advanced deployment options` section. If we consider setting up the deployment name and advanced option, why not the output? It is already separated from the `Output variables section` anyway.

> Introducing a new section just for the deployment name doesn't look good.

Correct.

> I would rather agree to the original change in the PR - with small change of making Deployment name input as last one in the main section. 

I would agree with that too if we ain't changing the section name. I did override the task in my account with exactly that change.

> Output is not a right fit for Advanced Deployment option. Lets stick to Advance Option.

If we change the section name, I think that's the better one for now.

So, what do we do guys?

Reply:
@andremarques023 , Pulkit and I discussed this and we feel the best choice right now is to rename `Outputs` section to `Advanced` section and have fields under it:

1. `Deployment name`
2. `Deployment outputs`

can you please update the PR accordingly?

Reply:
@bishal-pdMSFT sure, I'll have it fixed by this weekend.